### PR TITLE
Adding custom flags

### PR DESCRIPTION
**Pull request by taedium8vitae**
### Change-type: minor

Hey there ! I just added the variable `ENABLE_ZGC`, which if set to true will change the java startup flags to use [the Z Garbage Collector](https://wiki.openjdk.java.net/display/zgc/Main). I am asking to merge this change because it helped me reduce drastically RAM consumption on my server ! Like... DRASTICALLY !

### CODE REVIEW

File: mc-server/start.sh [hunk 1]
```diff
@@ -86,7 +86,12 @@ cd /usr/src/serverfiles/
 
 # Do that forever
 printf "%s\n" "Starting JAR file with: $RAM of RAM"
-java -Xms$RAM -Xmx$RAM -jar $JAR_FILE
```
Comment (taedium8vitae) [inline mc-server/start.sh:89]: I just rewrote it below, don't worry I didn't delete the most important part of the file...

**Review by AlexProgrammerDE — CHANGES_REQUESTED** [review]
The idea is good, but if we want to also add things like aikar flags and gc logging stuff or another garbage cleaner, it can quickly get overcomplicated. My idea would be to add a simple variable that combines all startup flags, can add predefined flags if set by a variable and allows the user to add custom flags too via a variable in the balena dashboard. Will soon try to work out a system for this.

**Comment by taedium8vitae** [comment]
The thing is, combining aikar's flags and the ZGC propably isn't a good idea. It really depends on what the user needs. Since this level of customisation will probably not be met by the majority of the users, I think two simple variables would be great. Example :
```
$ENABLE_FLAGS = true
$FLAGS = the user inserts his preferred flags here
```
The idea here is that the few people who want to go as far as customising the startup flags are probably willing to write them manually in a variable. This doesn't need to be "rookie friendly".
The only detail I haven't thought of is how we can implement the `$RAM` variable in this system, maybe make the Xms and Xmx stay even with the custom flags or just tell the user in the documentation how he can use it ?
Anyways, keep responding so we can figure the better way together !

**Comment by taedium8vitae** [comment]
PS : I converted it to draft so we can work on the project here if you want.

**Comment by AlexProgrammerDE** [comment]
Will make one last check and merge after that.

**Review by AlexProgrammerDE — APPROVED** [review]
LGTM! Thanks for your work!

**Pull request merged**